pilot/pkg/networking/core/listener_waypoint.go

```diff
@@ -326,13 +340,35 @@ func (lb *ListenerBuilder) buildWaypointInternal(wls []model.WorkloadInfo, svcs
 			}
 		}
 	}
+	// This may affect the data path due to the server-first protocols triggering a time-out.
+	// Currently, we attempt to exclude ports where we can, but it's not perfect.
+	// https://github.com/envoyproxy/envoy/issues/35958 is likely required for an optimal solution
+	httpInspector := xdsfilters.HTTPInspector
+	// If there are workloads, any port is accepted on them, so we cannot opt out
+	if len(wls) == 0 {
+		// Otherwise, find all the ports that have only TCP or only HTTP
+		nonInspectorPorts := []int{}
+		for p, proto := range portProtocols {
+			if !proto.IsUnsupported() {
+				nonInspectorPorts = append(nonInspectorPorts, p)
+			}
+		}
+		if len(nonInspectorPorts) > 0 {
+			// Sort for stable output, then replace the filter with one disabling the ports.
+			slices.Sort(nonInspectorPorts)
```

Comment:
To guarantee stable output should we use `SortStableFunc` instead?

Reply:
Its a set of distinct integrers, I believe the two functions are guaranteed to have the same output